ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java

```diff
@@ -52,23 +54,29 @@
   private final Offset offset;
   private final StoreKey storeKey;
   private final UUID sessionId;
+  private final UUID incarnationId;
   private long bytesRead;
 
   StoreFindToken() {
-    this(Type.Uninitialized, null, null, null);
+    this(Type.Uninitialized, null, null, null, null);
   }
 
-  StoreFindToken(StoreKey key, Offset indexSegmentStartOffset, UUID sessionId) {
-    this(Type.IndexBased, indexSegmentStartOffset, key, sessionId);
+  StoreFindToken(UUID incarnationId) {
```

Comment:
1. In findEntriesSince() in PersistentIndex, when we reset the token during validation, we return a StoreFindToken with just the incarnationId. 
2. During deserialization of StoreFindToken(VERSION_2), if the token is uninitialized, we create a StoreFindtoken with just the incarnationId that is read from the stream.